### ISSUE TITLE
Pull rep seqs for ASVs without taxonomy

### ISSUE THREAD

**Issue by aron1014** [benjjneb/dada2#1873]
I've been following your scripts to use different dada2 learnerror models to fix issues with novaseq quality scores (https://github.com/ErnakovichLab/dada2_ernakovichlab?tab=readme-ov-file#learn-the-error-rates). They work great by the way, thanks! However, for our dataset we are not assigning taxonomy, so I skipped those parts. By doing this, I am unable to extract rep sequences for ASVS as the code you provided uses the "taxonomy_for_fasta" dataframe. I've tried to use the writeRepSetFasta code for the "rep_set_ASVs" dataframe, but it does not work.

Ideally, I'd like to create a table (an export to excel) that has columns such as "ASV", "rep_sequence", Sample 1 reads, Sample 2 reads, Sample 3 reads, etc.

Basically, the seqtab_final.txt output with a column listing the sequences for each ASV. 

Any help is much appreciated, thanks!

Best,

Aron

**Comment by benjjneb** [comment]
You can manipulate the sequence table into whatever form you desire using R's data wrangling functionality. Here is something to should get you part of the way to where you are trying to go:

```
# Changes seqtab to whatever the variable name is of the sequence table you are trying to write out
seqtab.out <- as.data.frame(t(seqtab)) # Transpose the sequence table so rows are ASVs and samples are column
seqtab.out$rep_sequence <- rownames(seqtab.out)
write.csv(seqtab.out, row.names=FALSE, quote=FALSE)
```

This will produce a comma-separated-variable file with columns that are the abundances of ASVs in the samples and a final column that is the ASV sequence.

You can manipulate the data.frame more from there to get it exactly how you want (not sure what the "ASV" column is supposed to be, if not the ASV sequences?)

**Comment by aron1014** [comment]
Thank you! After I posted I had also figured out a different solution that outputs the rep_set_ASVs table which has the sequence info with each ASV name. I then just combined the read counts/sample with this.

write.table(rep_set_ASVs, file = paste0(table.fp, "/repseqs_final.txt"), 
            sep = "\t", row.names = TRUE, col.names = NA)


Thanks again for your quick response!
